--- dmd2/src/phobos/posix.mak.orig	2015-04-25 05:20:30.000000000 +0200
+++ dmd2/src/phobos/posix.mak	2015-07-22 15:28:25.126243307 +0200
@@ -160,7 +160,7 @@
 	PATHSEP:=$(shell echo "\\")
 endif
 
-LINKDL:=$(if $(findstring $(OS),linux),-L-ldl,)
+LINKDL:=$(if $(findstring $(OS),linux),-L-ldl -L-lz,)
 
 # Set VERSION, where the file is that contains the version string
 VERSION=../dmd/VERSION
@@ -262,7 +262,7 @@
 # Aggregate all C files over all OSs (this is for the zip file)
 ALL_C_FILES = $(C_FILES) $(C_EXTRAS)
 
-OBJS = $(addsuffix $(DOTOBJ),$(addprefix $(ROOT)/,$(C_MODULES)))
+OBJS =
 
 MAKEFILE = $(firstword $(MAKEFILE_LIST))
 
--- dmd2/tools/posix.mak.orig	2015-07-21 22:13:45.308848546 +0200
+++ dmd2/tools/posix.mak	2015-07-22 15:29:39.346240185 +0200
@@ -103,7 +103,7 @@
 	$(DMD) $(MODEL_FLAG) $(DFLAGS) -c -of$(@).o $(<)
 # grep for the linker invocation and append -lcurl
 	LINKCMD=$$($(DMD) $(MODEL_FLAG) $(DFLAGS) -v -of$(@) $(@).o 2>/dev/null | grep $(@).o); \
-	$${LINKCMD} -lcurl
+	$${LINKCMD} -lcurl -lz
 
 $(TOOLS) $(DOC_TOOLS): $(ROOT)/%: %.d
 	$(DMD) $(MODEL_FLAG) $(DFLAGS) -of$(@) $(<)
