--- dmd2/src/dmd/dmd/mars.d.orig	2024-06-30 18:32:38.000000000 +0200
+++ dmd2/src/dmd/dmd/mars.d	2024-12-02 19:55:00.354473035 +0100
@@ -357,11 +357,11 @@ void setDefaultLibrary(ref Param params,
             else
                 driverParams.defaultlibname = "phobos";
         }
-        else if (target.os & (Target.OS.linux | Target.OS.FreeBSD | Target.OS.OpenBSD | Target.OS.Solaris | Target.OS.DragonFlyBSD))
+        else if (target.os & (Target.OS.FreeBSD | Target.OS.OpenBSD | Target.OS.Solaris | Target.OS.DragonFlyBSD))
         {
             driverParams.defaultlibname = "libphobos2.a";
         }
-        else if (target.os == Target.OS.OSX)
+        else if (target.os & (Target.OS.linux | Target.OS.OSX))
         {
             driverParams.defaultlibname = "phobos2";
         }
